src/parser.rs

```diff
@@ -1180,7 +1183,7 @@ impl<'a> Parser<'a> {
             Token::Mul | Token::Div | Token::Mod | Token::StringConcat => Ok(40),
             Token::DoubleColon => Ok(50),
             Token::ExclamationMark => Ok(50),
-            Token::LBracket | Token::RBracket => Ok(10),
+            Token::LBracket => Ok(51),
```

Comment:
The value is bumped up because array index doesn't have right have operand. So, I bumped up to the highest value

Reply:
I know this was added before you, but it seems inappropriate to assign a precedence to a square bracket at all, instead of parsing it directly, like we do with [parens](https://github.com/sqlparser-rs/sqlparser-rs/blob/9569d1b21596c150643a212d09607aacd8f60677/src/parser.rs#L488).

Reply:
I had the same thought. I made it in a way, so that it can fit in the existing code. Now, I've removed the precedence.